### DISCUSSION TITLE
Best way to render a lot of FilamentInstances at same time

### DISCUSSION THREAD

**Discussion by ThomasGorisse** [google/filament#6763] [Q&A]
Hi.
I'm currently facing an issue while trying to render too much very simple model instances at same time.
As always, I solved a lot of surrounding things on SceneView but still got issues at rendering too much glTF at same time.
Good thing is that I don't see camera stream lags when not camera looking at models (object culling)
But as soon as too much glTFs are rendered at same time, even with latests versions of Filament and so the single call model multiple instances loading (without any loading time increase btw), I can see lags on the rendering.
Adding them to the scene and making all the math calculations on each ARCore frame update doesn’t generate any lags.
But as soon as objects become visible on the screen, lags are coming. Even with some improves made around the frame rendering vs transform order and should or shouldn't render the frame.
I also tried removing the alpha blending mode on the glTF (Don't know how to use something else on Blender than trasmission to fit my 0.5 alpha material needs but not sure it's optimum)
I tried a lot of Filament options and can’t find the right to disable any possible ray tracing that could generate too much calculations on reflections and transmissions.
My next try will be to remove everything on the model material in order to use an as simple as possible Unlit material (Don't know the corresponding on Blender). I will try to see if I need to go to defining the material programatically with a material less glTF. 
But I really thought so simple meshes (2 planes) would have been finger in the nose to render.

Do you have any advice for my case where I need to render very simple models (but no so simple to make them programmaticaly)?

**Comment by romainguy** [comment]
Can you share the glTF files being rendered, and explain how many instances you have on screen?

Filament doesn't have any ray tracing, but features like volume and transmission in glTF files can be expensive.

**Comment by ThomasGorisse** [comment]
The gltTF is made of 2 planes with this material:
<img width="666" alt="image" src="https://user-images.githubusercontent.com/6597529/234114507-5952adba-ced8-4bc7-80b8-a1fa2c2dbc72.png">
I firstly tryed with `Alpha Blend` but think it was worst.
I may display up to 100 same model instances within the camera view.

**Reply by romainguy** [reply]
Have you tried to use the profiler or perfetto to see where time is spent? Also please share the gift file so we can inspect it. Note: 100 blended objects will be expensive if they all cover a significant portion of the screen.

**Reply by ThomasGorisse** [reply]
<img width="1487" alt="image" src="https://user-images.githubusercontent.com/6597529/234248177-b4174d3f-71a1-4e3e-8c41-ae4d12995c23.png">
<img width="1487" alt="image" src="https://user-images.githubusercontent.com/6597529/234248686-5a097d34-dcf9-4065-910a-3b872be5607c.png">

Does that mean I made a big mistake ending in a `NavHost` recomposition on each surface frame callback (lol)?

Here is the gltf and .blend
[tile.zip](https://github.com/google/filament/files/11321509/tile.zip)

**Reply by romainguy** [reply]
Definitely not ideal but you should use the system profiler to get a better view of where time goes (CPU, GPU, etc.).

The glTF is strange, there are 2 planes on top of each other. That creates Z-fighting. Otherwise the material looks fine? If the planes are opaque, you're probably not GPU bound.
